app/Http/Controllers/SettingsController.php

```diff
@@ -3,19 +3,22 @@
 namespace App\Http\Controllers;
 
 use App\Settings;
+use App\Team;
 
 class SettingsController extends Controller
 {
     public function edit(Settings $setting)
-    {
-        return view('settings.edit', ['settings' => $setting]);
+    {   
+        $teams=Team::all();
+        return view('settings.edit', ['settings' => $setting,'teams'=>$teams]);
     }
 
     public function update(Settings $setting)
-    {
-        $this->validate(request(), [
-           'slack_webhook_url' => 'url',
-        ]);
+    {   
+        if(request()->slack_webhook_url!='')
+            $this->validate(request(), [
+            'slack_webhook_url' => 'url',
```

Comment:
I believe you can use the nullable flag here so you won't have to the the if

```
$this->validate(request(), [	 
    'slack_webhook_url' => 'nullable|url',
]);
```

Reply:
good idea, i'll change this